### PR TITLE
Add default kwargs logic into Anthropic provider, which is superseded by user entered kwargs

### PR DESCRIPTION
**Pull request by barisozmen**
A `DEFAULT_KWARGS` dictionary added to Anthropic provider class. Default max_tokens is inside of it. In the future, new default arguments can be added.

Before sending kwargs to the client, the default and user inputted kwargs are merged by the following pattern:
```
**{**self.DEFAULT_KWARGS, **kwargs}
```

## Summary by CodeRabbit

- **New Features**
	- Introduced a new constant, `DEFAULT_KWARGS`, to streamline handling of default arguments in key methods.
  
- **Improvements**
	- Enhanced flexibility of method signatures by merging default parameters with additional arguments in `send_conversation`, `structured_response`, and `generate_text` methods.

### CODE REVIEW

File: simplemind/providers/anthropic.py [hunk 1]
```diff
@@ -77,7 +80,7 @@ def generate_text(self, prompt: str, *, llm_model: str, **kwargs):
             model=llm_model or self.DEFAULT_MODEL,
             messages=messages,
             max_tokens=DEFAULT_MAX_TOKENS,
-            **kwargs,
+            **{**self.DEFAULT_KWARGS, **kwargs},
```
[the remark this answers is not in the document]
Comment (barisozmen) [reply simplemind/providers/anthropic.py]: Good idea

**Comment by kennethreitz** [comment]
✨🍰✨

**Comment by kennethreitz** [comment]
oh i merged too quickly

**Comment by barisozmen** [comment]
Sorry, does it break something?

Do you think it would be a good idea to implement the similar default_kwargs logic to other providers as well? I can do that at another PR

**Comment by kennethreitz** [comment]
yes i do.

**Comment by barisozmen** [comment]
I'm doing it

**Pull request merged**